File: HotelReservationServiceJava/RoomTypeC.java
//This room has both a minimum person and day number constaint
public class RoomTypeC extends Room
{

    int minimumPersonCount;
    int minimumDuration;
    
    public RoomTypeC(double perPersonPrice, int minimumPersonCount, int minimumDuration){
        super();
        this.perPersonPrice = perPersonPrice;
        this.minimumPersonCount = minimumPersonCount;
        this.minimumDuration = minimumDuration;
    }
    
    //checks constraints and propagates logic to Room.class
    @Override
    public boolean addReservation(Reservation res){
        if(res.getPersonCount() >= minimumPersonCount && res.getDuration() >= minimumDuration){
            return super.addReservation(res);
        }
        return false;
    }
    
}

File: HotelReservationServiceCpp/src/RoomTypeC.cpp
#include "RoomTypeC.h"

RoomTypeC::RoomTypeC(double ppPrice, int mpCount, int mDuration) :
    minimumPersonCount{mpCount},
    minimumDuration{mDuration},
    Room(mpCount , ppPrice)
{}

bool RoomTypeC::addReservation(Reservation& reservation){
    if(reservation.getPersonCount() >= minimumPersonCount && reservation.getDuration() >= minimumDuration){
        return Room::addReservation(reservation);
    }
    return false;
}

RoomTypeC::~RoomTypeC(){}
